android/engine/src/main/java/org/smartregister/fhircore/engine/ui/appsetting/AppSettingActivity.kt

```diff
@@ -45,42 +46,45 @@ class AppSettingActivity : AppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
     super.onCreate(savedInstanceState)
-    appSettingViewModel.loadConfigs.observe(
-      this,
-      { loadConfigs ->
-        if (loadConfigs != null && loadConfigs) {
-          val applicationId = appSettingViewModel.appId.value!!
-          configurationRegistry.loadAppConfigurations(
-            appId = applicationId,
-            accountAuthenticator = accountAuthenticator
-          ) { loadSuccessful: Boolean ->
-            if (loadSuccessful) {
-              sharedPreferencesHelper.write(APP_ID_CONFIG, applicationId)
-              finish()
-            } else {
-              showToast(
-                getString(R.string.application_not_supported, appSettingViewModel.appId.value)
-              )
-            }
+    appSettingViewModel.loadConfigs.observe(this) { loadConfigs ->
+      if (loadConfigs != null && loadConfigs) {
+        val applicationId = appSettingViewModel.appId.value!!
+        val compositionId = appSettingViewModel.compositionId.value!!
+        configurationRegistry.loadAppConfigurations(
+          appId = applicationId,
+          accountAuthenticator = accountAuthenticator
+        ) { loadSuccessful: Boolean ->
+          if (loadSuccessful) {
+            sharedPreferencesHelper.write(APP_ID_CONFIG, applicationId)
+            sharedPreferencesHelper.write(COMPOSITION_ID_CONFIG, compositionId)
```

Comment:
what is the functional difference btwn the app and composition ID? I assumed these would be equivalent, like if you supplied an app ID, it would know how to map it to a comp ID, why do we also need a comp ID?

Reply:
- This is true we could deprecate the app Id pieces once we fully have the composition id pieces working. Right now we need both for testing purposes until the pieces of the configurable app are fully done.  

- I think we can make this PR with a DO NOT MERGE until we have all the other underlying pieces complete and just couple the work here with deprecating the app id pieces.